backends/advanced/src/advanced_omi_backend/controllers/system_controller.py

```
@@ -1059,3 +966,187 @@ async def get_speaker_service_status():
+async def update_memory_config_raw(config_yaml: str):
+    """Update memory configuration YAML and hot reload."""
+    try:
+        import yaml
+        from advanced_omi_backend.memory_config_loader import get_config_loader
+        
+        # First validate YAML syntax
+        try:
+            yaml.safe_load(config_yaml)
+        except yaml.YAMLError as e:
+            return JSONResponse(
+                status_code=400, content={"error": f"Invalid YAML syntax: {str(e)}"}
```

Ruff (0.12.2)

Use explicit conversion flag

Replace with conversion flag

(RUF010)